### ISSUE TITLE
Keras_model_sequential(), no result : loss =NaN &  mae  = NaN

### ISSUE THREAD

**Issue by MARCMARITIME** [rstudio/keras#1221]
I tried to do a project of neural network on the same principle of the Boston housing project, but it failed at one point and I don't know how to solve it.

[test2.xlsx](https://github.com/rstudio/keras/files/6566971/test2.xlsx)

Here is my code:

`library(keras)
library(mlbench)  
library(psych)  
library(dplyr)
library(magrittr)
library(neuralnet)
library(readxl)
library(tidyverse)
library(broom)
library(tidyr)
library(MASS)
library(caret)
library(glmnet)
library(tensorflow)

# Data
data2 <- read_excel("C:\\Users\\mnicod\\test2.xlsx")
data2 %<>% mutate_if(is.factor,as.numeric)  
data2 %>% lapply(function(x) as.numeric(as.character(x)))
data2 <- data.frame(data2)

# Viz NN
n2 <- neuralnet(A0~A1+A2+A3+A4+A5+A6+A7+A8+A9+A10+A11+A12+A13+
                  A14+A15+A16+A17+A18+A19+A20+A21+A22+A23+A24+A25+A26+A27+A28+
                  A29+A30+A31+A32+A33+A34+A35+A36+A37+A38+A39+A40,
                data = data2,
                hidden = c(10,5),
                linear.output = F,
                lifesign = 'full',
                rep=1)
                
# Plot
plot(n2, col.hidden = "darkgreen", 
     col.hidden.synapse = 'darkgreen',
     show.weights = F, 
     information = F, 
     fill = "lightblue")
     
# Matrix
data2 <- as.matrix(data2)
dimnames(data2) <- NULL 

# Partition
set.seed(1234)
ind2 <- sample(2, nrow(data2), replace = T, prob=c(.7, .3))
training2 <- data2[ind2==1, c(2:41)]
test2 <- data2[ind2==2, 2:41]
trainingtarget2 <- data2[ind2==1, 1]
testtarget2 <- data2[ind2==2, 1]

# Normalize
m2 <- colMeans(training2)
sd2 <- apply(training2, 2, sd)
training2 <- scale(training2, center = m2, scale = sd2)
test2 <- scale(test2, center = m2, scale = sd2)

# Creat model
model2 <- keras_model_sequential()
model2 %>%  layer_dense(units = 10, activation = 'relu', input_shape = c(40)) %>%    
  layer_dense(units = 5, activation = 'relu') %>%
  layer_dense(units = 1)
  
  # Compile
  model2 %>% compile(loss = 'mse',  
                  optimizer = 'rmsprop',  
                  metrics = 'mae')
                  
# Fit the model
mymodel2 <- model2 %>%   
  fit(training2, 
      trainingtarget2, 
      epochs = 1000,
      batch_size = 32, #samples per gradient update default 32
      validation_split = 0.2)

console:
Epoch 998/1000
17/17 [==============================] - 0s 4ms/step - loss: nan - mae: nan - val_loss: nan - val_mae: nan
Epoch 999/1000
17/17 [==============================] - 0s 4ms/step - loss: nan - mae: nan - val_loss: nan - val_mae: nan
Epoch 1000/1000
17/17 [==============================] - 0s 4ms/step - loss: nan - mae: nan - val_loss: nan - val_mae: nan`      

# Evaluate 
model2 %>%  evaluate(test2, testtarget2)  

And then when I evaluate to the model, I didn't work:
9/9 [==============================] - 0s 999us/step - loss: nan - mae: nan
loss  mae 
 NaN  NaN
